### PR TITLE
feature/data-export-bus-19

### PR DESCRIPTION
**Pull request by pscheede**

### CODE REVIEW

File: Assembus/Assets/Scripts/Shared/DataExport.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        /// <summary>
+        ///     Export assembly data to a given file
+        /// </summary>
+        /// <param name="exportPath">Path to save the xml file to</param>
+        /// <returns>A boolean indicating the success and an error message in case of failure</returns>
+        public static (bool, string) ExportData(string exportPath)
+        {
+            // Save chosen file name
+            ProjectManager.CurrentProject.ExportFileName = Path.GetFileName(exportPath);
```
Comment (luismoser99) [inline Assembus/Assets/Scripts/Shared/DataExport.cs:22]: New export name doesn't get saved automatically to ProjectConfig!?

Comment (pscheede) [reply Assembus/Assets/Scripts/Shared/DataExport.cs:22]: @luismoser99 I'm sorry but I don't understand what you mean exactly.

Comment (luismoser99) [reply Assembus/Assets/Scripts/Shared/DataExport.cs:22]: I am required to additionally save the actual project after exporting the data in order to save my previous export-filename to the projectConfig file. Is this the intended behaviour?

File: Assembus/Assets/Scripts/Shared/DataExport.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                return (false, "Error while exporting data!");
+            }
+            return (true, "");
+        }
+
+        /// <summary>
+        ///     Convert the model data to the export XML format
+        /// </summary>
+        /// <returns>XElement representing the current internal data structure</returns>
+        private static XElement ConvertToExportXml()
```
Comment (luismoser99) [inline Assembus/Assets/Scripts/Shared/DataExport.cs:77]: The filename of the used OBJ-file is not saved in the exported XML file. Is this the intended behaviour?

File: Assembus/Assets/Scripts/Shared/DataExport.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            {
+                // Create new file for zip archive
+                using (var zipToSave = new FileStream(localExportPath, FileMode.Create))
+                {
+                    using (var archive = new ZipArchive(zipToSave, ZipArchiveMode.Update))
+                    {
+                        // Create entry in zip file for the xml export
+                        var xmlExportEntry = archive.CreateEntry(ProjectManager.CurrentProject.Name + ".xml");
+
+                        using (var writer = new StreamWriter(xmlExportEntry.Open()))
```
Comment (luismoser99) [inline Assembus/Assets/Scripts/Shared/DataExport.cs:37]: No Flush() call required?

**Review by luismoser99 — APPROVED** [review]
Very good work!

**Pull request merged**